design-library/src/components/BccTooltip/BccTooltip.css

```diff
@@ -0,0 +1,103 @@
+@layer components {
+  .bcc-tooltip {
+    @apply relative;
+  }
+
+  .bcc-tooltip-content {
+    @apply absolute left-1/2 text-center rounded-lg p-3 text-body-sm shadow-lg max-w-[200px] -translate-x-1/2 opacity-0 inline-block min-w-[80px] w-full;
+    transition: opacity 0.3s ease, visibility 0s linear 0.3s;
+  }
+
+  .bcc-tooltip-content.visible {
+    @apply opacity-100;
+    transition-delay: 0s;
+  }
+
+  .bcc-tooltip-content.top {
+    @apply bottom-full mb-2;
+  }
+  
+
+  .bcc-tooltip-content.bottom {
+    @apply top-full mt-2;
+  }
+
+  .bcc-tooltip-content.dark {
+    @apply bg-emphasis text-white;
+  }
+
+  .bcc-tooltip-content.white {
+    @apply bg-white text-primary;
+  }
+
+  .bcc-tooltip-content.grey {
+    @apply bg-neutral-100 text-secondary;
+  }
+
+  .bcc-tooltip-content.top-left {
+    @apply bottom-full mb-2 left-0 translate-x-0;
+  }
+
+  .bcc-tooltip-content.bottom-left {
+    @apply top-full mt-2 left-0 translate-x-0;
+  }
+
+  .bcc-tooltip-content.top-center {
+    @apply bottom-full mb-2 left-1/2;
+  }
+
+  .bcc-tooltip-content.bottom-center {
+    @apply top-full mt-2 left-1/2;
+  }
+
+  .bcc-tooltip-content.top-right {
+    @apply bottom-full mb-2 left-0 -translate-x-0;
+  }
+
+  .bcc-tooltip-content.bottom-right {
+    @apply top-full mt-2 left-0 -translate-x-0;
+  }
+
+  /* Arrow */
+  .bcc-tooltip-content::after {
+    content: "";
+    @apply absolute w-4 h-4 rounded rotate-45 -translate-x-1/2 left-1/2;
+  }
+
+  .bcc-tooltip-content.bottom-center::after {
+    @apply bottom-full -mb-[10px];
+  }
+
+  .bcc-tooltip-content.top-center::after {
+    @apply top-full -mt-[10px];
+  }
+
+  .bcc-tooltip-content.bottom-left::after {
+    @apply bottom-full -mb-[10px] left-[15%];
+  }
+
+  .bcc-tooltip-content.top-left::after {
+    @apply top-full -mt-[10px] left-[15%];
+  }
+
+  .bcc-tooltip-content.bottom-right::after {
+    @apply bottom-full -mb-[10px] left-[85%];
+  }
+
+  .bcc-tooltip-content.top-right::after {
+    @apply top-full -mt-[10px] left-[85%];
+  }
+
+  .bcc-tooltip-content.dark::after {
+    @apply bg-emphasis;
+  }
+
+  .bcc-tooltip-content.white::after {
+    @apply bg-white;
+  }
+
+  .bcc-tooltip-content.grey::after {
+    @apply bg-neutral-100;
+  } 
```

Comment:
I think it's best if we prefix all these modifier classes with `bcc-tooltip` as well. It makes this CSS look a bit more ugly, but I feel a class like `white` can easily clash with some projects. Same for `visible`, `top` etc. See the button component for an example of how we match props to class names (a simple key-value object map).

Reply:
you are right! I will change this